src/cavendish_particle_tracks/_decay_angles_dialog.py

```diff
@@ -55,30 +53,83 @@ def __init__(self, parent=None):
+
+        # # This almost gets the behavior we want, but only for the line labels
+        # def enforce_points_coincident_v2(event: Event) -> None:
+        #     (selected_shape,) = self.cal_layer.selected_data
+        #     data = self.cal_layer.data
+        #     if len(data) != 3:
+        #         return
+        #     if selected_shape==0 and (data[0][0] != data[1][0]).all():
+        #         # # This is what we want but needs private methods
+        #         # vertices = self.cal_layer._data_view.shapes[1].data
+        #         # vertices[0] = data[0][0]
+        #         # self.cal_layer._data_view.edit(1, vertices, new_type=None)
+        #         # self.cal_layer.refresh()
+        #         data[1][0][0] = data[0][0][0]
+        #         data[1][0][1] = data[0][0][1]
+        #         self.cal_layer.refresh()
+        # self.cal_layer.events.set_data.connect(enforce_points_coincident_v2)
 
         # Decay Angles related parameters
         self.phi_proton = 0.0
         self.phi_pion = 0.0
         self.alines = []
 
+    def _enforce_points_coincident(self, event: Event) -> None:
+        # event.data_indices says what shape was changed
+        # action says what was done to the shape
+        # data doesn't get set till user lets go, mouse drag might be better, but not sure how to implement it
+        # Maybe the long term solution is to have a custom shape?
+        if event.action == "changed":
+            shapes_modified = event.data_indices
+            data = self.cal_layer.data
+            if len(shapes_modified) == 3:
+                # if all shapes were modified, this must be a translation of the three shapes
+                # do nothing.
+                return
+            elif len(shapes_modified) == 2:
+                # if two of them have been moved, move the third one (if it has not been moved already)
+                for i in range(3):
+                    if (i not in shapes_modified) and (
+                        self.cal_layer.data[i][0]
+                        != self.cal_layer.data[shapes_modified[0]][0]
+                    ).any():
+                        data[i][0] = self.cal_layer.data[shapes_modified[0]][0]
+                        self.cal_layer.data = data
+            elif len(shapes_modified) == 1:
+                # if only one has been moved, move the other two (if they have not been moved already)
+                for i in range(3):
+                    if (i not in shapes_modified) and (
+                        self.cal_layer.data[i][0]
+                        != self.cal_layer.data[shapes_modified[0]][0]
+                    ).any():
+                        data[i][0] = self.cal_layer.data[shapes_modified[0]][0]
+                        self.cal_layer.data = data
+            else:
+                # nothing moved, or more shapes have been added
+                # do nothing
+                return
```

Comment:
Flip the logic to _guard_ the active code. Then you can unindent (#90CharactersPerLine).

```py
if event_action is not "changed":
    return  # Nothing has been moved, so nothing to do

shapes_modified = event.data_indices
data = self.cal_layer.data

...
```